src/univariate/continuous/generalizedpareto.jl

```diff
@@ -169,6 +169,189 @@ function quantile(d::GeneralizedPareto{T}, p::Real) where T<:Real
     return μ + σ * z
 end
 
+#### Fitting
+
+#
+# MLE
+#
+
+"""
+    GeneralizedParetoKnownMuTheta(μ, θ)
+
+Represents a [`GeneralizedPareto`](@ref) where ``\\mu`` and ``\\theta=\\frac{\\xi}{\\sigma}`` are known.
+"""
+struct GeneralizedParetoKnownMuTheta{T} <: IncompleteDistribution
+    μ::T
+    θ::T
+end
+GeneralizedParetoKnownMuTheta(μ, θ) = GeneralizedParetoKnownMuTheta(Base.promote(μ, θ)...)
+
+struct GeneralizedParetoKnownMuThetaStats{T} <: SufficientStats
+    μ::T  # known mean
+    θ::T  # known theta
+    ξ::T  # known shape
+end
+function GeneralizedParetoKnownMuThetaStats(μ, θ, ξ)
+    return GeneralizedParetoKnownMuThetaStats(Base.promote(μ, θ, ξ)...)
+end
+
+function suffstats(d::GeneralizedParetoKnownMuTheta, x::AbstractArray)
+    μ = d.μ
+    θ = d.θ
+    ξ = mean(xi -> log1p(θ * (xi - μ)), x) # mle estimate of ξ
+    return GeneralizedParetoKnownMuThetaStats(μ, θ, ξ)
+end
+
+"""
+    fit_mle(GeneralizedPareto, x; μ, θ)
+
+Compute the maximum likelihood estimate of the parameters of a [`GeneralizedPareto`](@ref)
+where ``\\mu`` and ``\\theta=\\frac{\\xi}{\\sigma}`` are known.
+"""
+function fit_mle(::Type{<:GeneralizedPareto}, x::AbstractArray; μ::Real, θ::Real)
+    g = GeneralizedParetoKnownMuTheta(μ, θ)
+    ss = suffstats(g, x)
+    return fit_mle(g, ss)
+end
+function fit_mle(d::GeneralizedParetoKnownMuTheta, ss::GeneralizedParetoKnownMuThetaStats)
+    ξ = ss.ξ
+    return GeneralizedPareto(d.μ, ξ / d.θ, ξ)
+end
+
+#
+# empirical bayes
+#
+
+"""
+    GeneralizedParetoKnownMu(μ)
+
+Represents a [`GeneralizedPareto`](@ref) where ``\\mu`` is known.
+"""
+struct GeneralizedParetoKnownMu{T} <: IncompleteDistribution
+    μ::T
+end
+
+"""
+    fit(GeneralizedPareto, x; μ, kwargs...)
+
+Fit a [`GeneralizedPareto`](@ref) with known location `μ` to the data `x`.
+
+The fit is performed using the Empirical Bayes method of [^ZhangStephens2009][^Zhang2010].
```

Comment:
From the linked page
> The fit function will choose a reasonable way to fit the distribution, which, in most cases, is maximum likelihood estimation.
This doesn't guarantee that only MLE or MAP are used, just a reasonable method.

While I don't know of other non-MLE/non-MAP examples in this package, for GPD, the MLE would indeed require reimplementing a Newton method and is not ideal since the likelihood is infinite for certain parameter values. As covered in the linked papers, the method here is less biased, more accurate, and more statistically efficient than the MLE, while also being simpler to compute. As such, I'd argue this is a more reasonable and convenient fitting method and is consistent with the documentation.

Reply:
> From the linked page
> > The fit function will choose a reasonable way to fit the distribution, which, in most cases, is maximum likelihood estimation.
> 
> This doesn't guarantee that only MLE or MAP are used, just a reasonable method.

I read this sentence but my interpretation was a bit different. Since the documentation talks only about MLE and MAP (and I think only these are supported currently) I just assumed that usually `fit` would call `fit_mle` but sometimes it might also use `fit_map`.

> While I don't know of other non-MLE/non-MAP examples in this package, for GPD, the MLE would indeed require reimplementing a Newton method

This is what's currently done in Distributions for other distributions though, e.g., https://github.com/JuliaStats/Distributions.jl/blob/837d3128bfbb30a0aa00e7372b99313f30da635f/src/multivariate/dirichlet.jl#L278

To be clear: I'm not strongly against adding `fit_empiricalbayes` but I think it requires some additional thought and discussion since 1) currently such estimation procedures are not part of Distributions and 2) it is not clear to me from the documentation if it was/is the intention to support only MLE and MAP.

Reply:
> To be clear: I'm not strongly against adding `fit_empiricalbayes` but I think it requires some additional thought and discussion since 1) currently such estimation procedures are not part of Distributions and 2) it is not clear to me from the documentation if it was/is the intention to support only MLE and MAP.

I agree, this needs to be discussed, and is I think the main question, because if this doesn't belong here, then this PR is not useful. To be clear, `fit_empiricalbayes` is not an exported method and is not intended to be used directly.

Reply:
Personally, I think that it should be added but that it could still be useful at some point to move the `fit` etc. implementations to a separate package (due to the reasons mentioned above). IMO the main reason for adding it is that the method seems simpler and apparently is less biased and more efficient than MLE.

We should update the documentation though and clearly describe which estimates are used. The list of MLE estimates should be updated but I think we should also add a section with non-MLE estimates.

What do you think, @matbesancon @mschauer @andreasnoack?